perfkitbenchmarker/benchmarks/cassandra_stress_benchmark.py

```diff
@@ -188,36 +191,29 @@ def WaitForLoaderToFinish(vm):
     time.sleep(SLEEP_BETWEEN_CHECK_IN_SECONDS)
 
 
-def CollectResultFile(vm, interval_op_rate_list, interval_key_rate_list,
-                      latency_median_list, latency_95th_list,
-                      latency_99_9th_list,
-                      total_operation_time_list):
+def CollectResultFile(vm, results):
   """Collect result file on vm.
 
   Args:
     vm: The target vm.
-    interval_op_rate_list: The list stores interval_op_rate.
-    interval_key_rate_list: The list stores interval_key_rate.
-    latency_median_list: The list stores latency median.
-    latency_95th_list: The list stores latency 95th percentile.
-    latency_99_9th_list: The list stores latency 99.9th percentile.
-    total_operation_time_list: The list stores total operation time.
+    results: A dictionary of lists. Each list contains results of a field defined in
+        RESULTS_METRICS collected from each loader machines.
   """
   result_path = _ResultFilePath(vm)
   vm.PullFile(vm_util.GetTempDir(), result_path)
-  resp, _ = vm.RemoteCommand('tail ' + result_path)
-  match = re.findall(r'[\w\t ]: +([\d\.:]+)', resp)
-  if len(match) < 6:
-    raise ValueError('Result not found in "%s"' % resp)
-  interval_op_rate_list.append(int(match[0]))
-  interval_key_rate_list.append(int(match[1]))
-  latency_median_list.append(float(match[2]))
-  latency_95th_list.append(float(match[3]))
-  latency_99_9th_list.append(float(match[4]))
-  raw_time_data = match[5].split(':')
-  total_operation_time_list.append(
-      int(raw_time_data[0]) * 3600 + int(raw_time_data[1]) * 60 + int(
-          raw_time_data[2]))
+  resp, _ = vm.RemoteCommand('tail -n 20 ' + result_path)
+  for metric in RESULTS_METRICS:
+
+    try:
+      value = regex_util.ExtractGroup(r'%s[\t ]+: ([\d\.:]+)' % metric, resp)
+      if metric == RESULTS_METRICS[-1]:  # Total operation time
+        value = value.split(':')
+        results[metric].append(
+            int(value[0]) * 3600 + int(value[1]) * 60 + int(value[2]))
+      else:
+        results[metric].append(float(value))
+    except regex_util.NoMatchError:
+      logging.exception('No value for %s', metric)
```

Comment:
This doesn't occur in current metrics. 
But the new cassandra stress tool also reports:
"Total partitions", "Total errors", "total gc count", "total gc mb", "total gc time (s)", "avg gc time (ms)", "stdev gc time(ms)". The avg gc time might see "NaN" if no gc occurs during the test. I haven't added these new metrics yet.